src/Nancy.Tests/Unit/Configuration/DefaultNancyEnvironmentFixture.cs

```diff
@@ -0,0 +1,192 @@
+        [InlineData(6)]
+        public void Should_add_value_to_environment_when_invoking_addvalue(int numberOfElementsToAdd)
+        {
+            // Given, When
+            var environment = CreateEnvironment(numberOfElementsToAdd);
+            IReadOnlyDictionary<string, object> setter = environment;
+
+            // Then
+            for (var value = 0; value < numberOfElementsToAdd; value++)
+            {
+                setter[value.ToString()].Equals(value).ShouldBeTrue();
+            }
+        }
+
+        [Theory]
+        [InlineData(7)]
+        [InlineData(9)]
+        [InlineData(12)]
+        public void Should_return_all_keys_when_invoking_keys(int numberOfElementsToAdd)
+        {
+            // Given,
+            var environment = CreateEnvironment(numberOfElementsToAdd);
+
+            // When
+            var returnedKeys = environment.Keys.ToArray();
+
+            // Then
+            returnedKeys.Count().ShouldEqual(numberOfElementsToAdd);
+
+            for (var value = 0; value < numberOfElementsToAdd; value++)
+            {
+                returnedKeys.Contains(value.ToString()).ShouldBeTrue();
+            }
+        }
+
+        [Theory]
+        [InlineData(7)]
+        [InlineData(9)]
+        [InlineData(12)]
+        public void Should_return_all_values_when_invoking_values(int numberOfElementsToAdd)
+        {
+            // Given,
+            var environment = CreateEnvironment(numberOfElementsToAdd);
+
+            // When
+            var returnedValues = environment.Values.ToArray();
+
+            // Then
+            returnedValues.Count().ShouldEqual(numberOfElementsToAdd);
+
+            for (var value = 0; value < numberOfElementsToAdd; value++)
+            {
+                returnedValues.Contains(value).ShouldBeTrue();
+            }
+        }
+
+        [Theory]
+        [InlineData(7)]
+        [InlineData(9)]
+        [InlineData(12)]
+        public void Should_return_count_when_invoking_count(int numberOfElementsToAdd)
+        {
+            // Given
+            var environment = CreateEnvironment(numberOfElementsToAdd);
+
+            // When
+            var returnedCount = environment.Count;
+
+            // Then
+            returnedCount.ShouldEqual(numberOfElementsToAdd);
+        }
+
+        [Theory]
+        [InlineData("nancy", "nancy", true)]
+        [InlineData("nancy", "frank", false)]
+        public void Should_return_correct_key_presence_when_invoking_containskey(string keyToAdd, string keyToLookFor, bool expectedResult)
+        {
+            // Given
+            var environment = new DefaultNancyEnvironment();
+            environment.AddValue(keyToAdd, new object());
+
+            // When
+            var result = environment.ContainsKey(keyToLookFor);
+
+            // Then
+            result.ShouldEqual(expectedResult);
+        }
+
+        [Fact]
+        public void Should_return_true_if_key_exists_when_invoking_trygetvalue()

```

Comment:
This test and `Should_return_false_if_key_does_not_exist_when_invoking_trygetvalue()` could be unified with some `InilineData` like above.